stream/hls.go

```diff
@@ -25,18 +31,19 @@ type HLSMuxer interface {
 
 //TODO: Write tests, set buffer size, kick out segments / playlists if too full
 type HLSBuffer struct {
-	HoldTime   time.Duration
 	plCacheNew bool
-	segCache   *Queue
+	segCache   []string
 	// pq       *Queue
-	plCache m3u8.MediaPlaylist
-	sq      *ConcurrentMap
-	lock    sync.Locker
+	plCache  m3u8.MediaPlaylist
+	sq       *ConcurrentMap
+	lock     sync.Locker
+	Capacity uint
 }
 
-func NewHLSBuffer() *HLSBuffer {
```

Comment:
Should we just leave this method and invoke the new method with a default capacity?